lib/controllers/fhir_questionnaire/questionnaire_controller.dart

```diff
@@ -13,52 +12,49 @@ class QuestionnaireController extends GetxController {
   final QuestionnaireModel _model = QuestionnaireModel();
 
   final UserResponsesController _responsesController = Get.find();
-  List<Question> _allQuestions;
+  List<SurveyItem> _allQuestions;
 
   //todo: implement error handling / orElse
-  Survey getSurveyFromCode(String code) => _model.data.surveys
-      .firstWhere((e) => e.code == code, orElse: () => Survey());
+  ItemGroup getGroupFromCode(String code) => _model.data.survey.surveyItems
+      .firstWhere((e) => e.linkId == code, orElse: () => ItemGroup());
 
   //todo: implement error handling / orElse
-  Survey getSurveyFromIndex(int sIndex) => _model.data.surveys[sIndex];
+  ItemGroup getSurveyFromIndex(int sIndex) =>
+      _model.data.survey.surveyItems[sIndex];
 
   //todo: implement error handling / orElse
-  int getSurveyIndexFromSurvey(Survey survey) =>
-      _model.data.surveys.indexWhere((e) => e == survey);
+  int getSurveyIndexFromSurvey(ItemGroup itemGroup) =>
+      _model.data.survey.surveyItems.indexWhere((e) => e == itemGroup);
 
   FhirQuestionnaire getQuestionnaire() => _model.data;
 
   int getTotalIndexFromQuestion(Question question) =>
       _allQuestions.indexOf(question);
 
   void _mapAllQuestions() {
-    _allQuestions =
-        _model.data.surveys.map((e) => e.questions).expand((x) => x).toList();
+    _allQuestions = _model.data.survey.surveyItems
+        .map((e) => (e as ItemGroup).surveyItems)
+        .expand((x) => x)
+        .toList();
   }
 
   void _mapAllUserResponses() {
-    _model.data.surveys.forEach(
-      (s) => s.questions.forEach(
-        (q) => q.answers.forEach(
-          (ans) => _responsesController.rxResponses.add(
-            // if AnswerType.choice ==> use booleans, otherwise strings
-            // todo: update with various additional response types
-            (ans.code == 'decimal' || ans.code == 'string')
-                ? UserResponse(
-                        surveyCode: s.code,
-                        questionCode: q.code,
-                        answerCode: ans.code,
-                        responseType: ResponseString(''))
-                    .obs
-                : UserResponse(
-                        surveyCode: s.code,
-                        questionCode: q.code,
-                        answerCode: ans.code,
-                        responseType: ResponseBoolean(false))
-                    .obs,
+    _model.data.survey.surveyItems.forEach(
+      (s) => (s as ItemGroup).surveyItems.forEach(
+            (q) => (q as Question).answers.forEach(
+                  (ans) => _responsesController.rxResponses.add(
+                    // if AnswerType.choice ==> use booleans, otherwise strings
+                    // todo: update with various additional response types
+                    (ans.code == 'decimal' || ans.code == 'string')
+                        ? UserResponse(
```

Comment:
This can be a switch statement now

Reply:
If we want to add different answer types, we can do that here in a switch statement.